### ISSUE TITLE
Unexpected GraphQL Error in UI

### ISSUE THREAD

**Issue by aleexharris** [dagster-io/dagster#25390]
### What's the issue?

When I open the details of a specific Run, I get a red pop-up in the corner which says "Unexpected GraphQL error, View error info"

When I view the error info it says:

```
Operation name: {PipelineRunLogsSubscription}

Message: FSEventsObserver.schedule() takes 3 positional arguments but 4 were given

Path: 

Locations: 
```

This is whilst doing the Dagster Essentials course on Dagster University.

### What did you expect to happen?

Your UI told me to report the bug.

### How to reproduce?

Every time I open the details of a Run, I get this pop-up in the top right-hand corner.

### Dagster version

dagster, version 1.7.16

### Deployment type

None

### Deployment details

`dagster dev` on local machine in iTerm using zsh

### Additional information

My machine is a 13-inch, M1 Macbook Pro from 2020 running MacOS Sonoma 14.3.1. I'm on Python 3.12.1 running a virtual env using `pyenv` and `pyenv-virtualenv`. My pip freeze is:
```
alembic==1.13.3
annotated-types==0.7.0
anyio==4.6.2.post1
backoff==2.2.1
certifi==2024.8.30
cffi==1.17.1
charset-normalizer==3.4.0
click==8.1.7
coloredlogs==14.0
croniter==3.0.3
cryptography==43.0.3
dagster==1.7.16
dagster-cloud==1.7.16
dagster-cloud-cli==1.7.16
dagster-duckdb==0.23.16
dagster-graphql==1.7.16
dagster-pipes==1.7.16
dagster-webserver==1.7.16
# Editable Git install with no remote (dagster_university==0.0.0)
-e /Users/alex/projects/dagster-essentials/dagster_university
docstring_parser==0.16
duckdb==1.1.2
filelock==3.16.1
fsspec==2024.10.0
geopandas==1.0.1
github3.py==4.0.1
gql==3.5.0
graphene==3.4
graphql-core==3.2.5
graphql-relay==3.2.0
grpcio==1.64.3
grpcio-health-checking==1.62.3
h11==0.14.0
httptools==0.6.4
humanfriendly==10.0
idna==3.10
iniconfig==2.0.0
Jinja2==3.1.4
kaleido==0.2.1
Mako==1.3.5
markdown-it-py==3.0.0
MarkupSafe==3.0.2
mdurl==0.1.2
multidict==6.1.0
numpy==2.1.2
packaging==24.1
pandas==2.2.3
pendulum==3.0.0
pex==2.20.3
plotly==5.24.1
pluggy==1.5.0
prompt-toolkit==3.0.36
propcache==0.2.0
protobuf==4.25.5
pycparser==2.22
pydantic==2.9.2
pydantic_core==2.23.4
Pygments==2.18.0
PyJWT==2.9.0
pyogrio==0.10.0
pyproj==3.7.0
pytest==8.3.3
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
pytz==2024.2
PyYAML==6.0.2
questionary==2.0.1
requests==2.32.3
requests-toolbelt==1.0.0
rich==13.9.2
setuptools==75.2.0
shapely==2.0.6
shellingham==1.5.4
six==1.16.0
sniffio==1.3.1
SQLAlchemy==2.0.36
starlette==0.41.0
structlog==24.4.0
tabulate==0.9.0
tenacity==9.0.0
time-machine==2.16.0
tomli==2.0.2
toposort==1.10
tqdm==4.66.5
typer==0.12.5
typing_extensions==4.12.2
tzdata==2024.2
universal_pathlib==0.2.5
uritemplate==4.1.1
urllib3==2.2.3
uvicorn==0.32.0
uvloop==0.21.0
watchdog==5.0.3
watchfiles==0.24.0
wcwidth==0.2.13
websockets==13.1
yarl==1.15.5
```

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.
By submitting this issue, you agree to follow Dagster's [Code of Conduct](https://github.com/dagster-io/dagster/blob/master/.github/CODE_OF_CONDUCT.md).

**Comment by elliottower** [comment]
I've been getting this as well, just following the dagster university tutorials

**Comment by garethbrickman** [comment]
Duplicates https://github.com/dagster-io/dagster/issues/23957

This a known issue with the ​​watchdog​​ dependency due to a breaking change the library introduced. It is fixed in dagster versions =>1.8.4.

Adding a ​​watchdog<5​​ pin would also fix it in earlier versions.
